packages/sanity/src/_internal/cli/actions/app/devAction.ts

```diff
@@ -2,14 +2,33 @@ import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
 
 import {startDevServer} from '../../server/devServer'
 import {gracefulServerDeath} from '../../util/servers'
-import {getDevServerConfig, type StartDevServerCommandFlags} from '../dev/devAction'
+import {getCoreAppURL, getDevServerConfig, type StartDevServerCommandFlags} from '../dev/devAction'
 
 export default async function startAppDevServer(
   args: CliCommandArguments<StartDevServerCommandFlags>,
   context: CliCommandContext,
 ): Promise<void> {
   const flags = args.extOptions
+
   const {output, workDir, cliConfig} = context
+  if (flags.loadInDashboard === false) {
+    output.warn(`Core applications cannot be run without dashboard`)
+    output.warn(`Starting dev server with the --loadInDashboard flag set to true`)
+  }
+
+  let organizationId: string | undefined
+  if (
+    cliConfig &&
+    '__experimental_coreAppConfiguration' in cliConfig &&
+    cliConfig.__experimental_coreAppConfiguration?.organizationId
+  ) {
+    organizationId = cliConfig.__experimental_coreAppConfiguration.organizationId
+  }
+
+  if (!organizationId) {
+    output.error(`Core applications require an organization ID`)
```

Comment:
Think it might be helpful to list out instructions on how to find your org ID so added those. 

```
output.error(`Apps require an organization ID (orgId) specified in your sanity.cli.ts file. \\n

To find and add your orgId: \\n
1. Go to https://sanity.io/manage. \\n
2. Select your organization from the list. \\n
3. Look at the URL and copy the value after "?orgId=". \\n
   For example, in this URL:
   sanity.io/manage/organizations/[id]/projects/[id]/?orgId=o5RnUYXOb
   Your orgId would be o5RnUYXOb
4. Open your sanity.cli.ts file and add this line with your orgID:
   organizationId: 'YOUR_ORG_ID_HERE',
`)
```

Reply:
I am reluctant to add specific links and instructions as things might change and CLI code "lives forever" this might become more confusing then helpful

Reply:
Hmmm an interesting constraint but curious how we balance that with providing clear instructions on how to keep going. I really don't want to leave users with errors where they don't understand clearly how to move forward and resolve the issue.